### ISSUE TITLE
a11y: Fix keyboard navigation, aria label/role and focus display for user menu

### ISSUE THREAD

**Issue by nwmac** [rancher/dashboard#12790]
a11y Issues: 2, 9, 10, 33

Sub-menu links get displayed automatically when the "Profile" interactive image receives the keyboard focus resulting in keyboard-only and screen reader users will have to navigate through the sub-menu links even when not required.

![image](https://github.com/user-attachments/assets/853e36d7-9df9-4018-a462-c6615bf8a736)

Should fix by using a common menu component.

The images used for the logo and the avatar and missing the `alt` attribute or the `aria-label` attributes:

![image](https://github.com/user-attachments/assets/90938247-58a2-4190-b0c7-4eb333b10bde)

![image](https://github.com/user-attachments/assets/53d3bd21-80a2-409d-a705-16feaa549ac5)

(Issue 9 is the logo, Issue 10 is the profile/avatar image)

Issue 33: The "alt" attribute is not provided for the informative images, resulting in screen reader announcing the images as unlabelled to their users.

![image](https://github.com/user-attachments/assets/7c12e167-f8a2-465a-8fb6-f83604fa0f69)

^^ Note this covers both the user avatar icon AND the other header icon (including here for simplicity)

**Comment by yonasberhe23** [comment]
Requires manual validation.

**Comment by izaac** [comment]
Rancher version: `v2.11-0b081e077e830fa17b86151d21c786fb8d07c6de-head`
UI version: `master 68fb044`
Chrome version: `133.0.6943.141`
Firefox version: `135.0.1`

This is working as expected in this Rancher/UI build. ✅
